File: CSharp/Easy/1-two-sum/two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> map = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++){
            int  complement_number = target - nums[i];

            if (map.ContainsKey(complement_number)){
                return new int[] {map[complement_number], i};
            }

            if (!map.ContainsKey(nums[i])){
                map.Add(nums[i], i);
            }
        }
        return new int[0];
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        previous_map = {}

        for i, n in enumerate(nums):
            complement_number = target - n

            if complement_number in previous_map:
                return [previous_map[complement_number], i]

            previous_map[n] = i

        return []